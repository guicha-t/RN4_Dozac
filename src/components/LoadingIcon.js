import * as React from 'react';
import { ActivityIndicator, View } from 'react-native';

export default function LoadingIcon() {
  return (
    <View style={{ flex: 1, justifyContent: 'center' }}>
      <ActivityIndicator size="large" color="#0000ff" />
    </View>
  );
}
