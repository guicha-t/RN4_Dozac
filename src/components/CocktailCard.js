import React, { useRef, useState } from 'react';

import { View, TouchableHighlight, Text, Image, Dimensions, StyleSheet } from 'react-native';
import { Button, Card } from 'react-native-elements';
import { TouchableOpacity } from 'react-native-gesture-handler';
import StyleWrapper from '../HOC/styleHOC';

import LoadingIcon from './LoadingIcon';

import { globalStyle } from './GeneralStyles';
// import { TouchableHighlight } from 'react-native-gesture-handler';

const styles = StyleSheet.create({});

export default function CocktailCard({ name, pic, cocktail, onPress, sty }) {
  if (cocktail === undefined) {
    return <LoadingIcon />;
  }
  return (
    <View style={sty}>
      <TouchableOpacity
        style={{ justifyContent: 'center', alignItems: 'center' }}
        onPress={onPress}
      >
        <Image
          style={{ height: 70, width: 70, margin: 3 }}
          source={
            pic
              ? {
                  uri: `${pic}/preview`,
                }
              : require('../../assets/broken.png')
          }
          resizeMode="contain"
        />
        <View>
          <Text style={{ flexWrap: 'wrap', textAlign: 'center', color: 'brown' }}>{name}</Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}
