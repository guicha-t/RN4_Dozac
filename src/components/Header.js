import React from 'react';

import { View, TouchableOpacity, Text, Image, Dimensions, StyleSheet } from 'react-native';
import { Button, Icon, Divider, withBadge } from 'react-native-elements';
import { LinearGradient } from 'expo-linear-gradient';

import { TranslateRouter } from './translation';
import { headerStyle } from './GeneralStyles';
import LoadingIcon from './LoadingIcon';

const MessagesBadge = withBadge(5)(Icon);
const MessagesIcon = () => <MessagesBadge type="font-awesome" name="bell" size={24} color="#fff" />;

export default function Header({ scene, previous, navigation }) {
  return (
    <View style={headerStyle.container}>
      {previous ? (
        <Button
          type="clear"
          icon={<Icon name="chevron-left" type="font-awesome" color="#fff" />}
          onPress={navigation.goBack}
        />
      ) : (
        <Button
          type="clear"
          icon={<MessagesIcon name="chevron-left" type="font-awesome" color="#fff" />}
          onPress={navigation.goBack}
        />
      )}
      <Image
        style={{ height: 30, width: 30 }}
        source={require('../../assets/mojito.png')}
        resizeMode="contain"
      />
      <Image
        style={{ height: 30, width: 30 }}
        source={require('../../assets/mojito.png')}
        resizeMode="contain"
      />
    </View>
  );
}
