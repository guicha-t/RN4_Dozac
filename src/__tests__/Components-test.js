import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import CocktailCard from '../components/CocktailCard';
import Header from '../components/Header';
import LoadingIcon from '../components/LoadingIcon';

it('CocktailCard - renders correctly', () => {
  const tree = renderer.create(<CocktailCard />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Header - renders correctly', () => {
  const tree = renderer.create(<Header />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('LoadingIcon - renders correctly', () => {
  const tree = renderer.create(<LoadingIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});
