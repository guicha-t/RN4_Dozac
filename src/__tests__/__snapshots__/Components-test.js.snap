// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CocktailCard - renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#0000ff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;

exports[`LoadingIcon - renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#0000ff"
    hidesWhenStopped={true}
    size="large"
  />
</View>
`;
